src/couch_views/test/couch_views_indexer_test.erl

```diff
@@ -372,6 +376,76 @@ index_autoupdater_callback(Db) ->
     ?assertEqual(ok, couch_views_jobs:wait_for_job(JobId, DbSeq)).
 
 
+handle_db_recreated_when_running(Db) ->
+    DbName = fabric2_db:name(Db),
+
+    DDoc = create_ddoc(),
+    {ok, _} = fabric2_db:update_doc(Db, DDoc, []),
+    {ok, _} = fabric2_db:update_doc(Db, doc(0), []),
+    {ok, _} = fabric2_db:update_doc(Db, doc(1), []),
+
+    % To intercept job building while it is running ensure updates happen one
+    % row at a time.
+    config:set("couch_views", "change_limit", "1", false),
+
+    meck_intercept_job_update(self()),
+
+    [{ok, JobId}] = couch_views:build_indices(Db, [DDoc]),
+
+    {Indexer, _Job, _Data} = wait_indexer_update(10000),
+
+    {ok, State} = couch_jobs:get_job_state(undefined, ?INDEX_JOB_TYPE, JobId),
+    ?assertEqual(running, State),
+
+    ok = fabric2_db:delete(DbName, []),
+    {ok, Db1} = fabric2_db:create(DbName, [?ADMIN_CTX]),
+
+    {ok, _} = fabric2_db:update_doc(Db1, DDoc, []),
+    {ok, _} = fabric2_db:update_doc(Db1, doc(2), []),
+    {ok, _} = fabric2_db:update_doc(Db1, doc(3), []),
+
+    reset_intercept_job_update(Indexer),
+
+    {ok, Out2} = run_query(Db1, DDoc, ?MAP_FUN1),
+    ?assertEqual([
```

Comment:
These assertions seem a bit vague. I realize you've checked that the docs don't exist before the db recreation, but that would have deleted the entire view as well. Given that we're testing that the job failed, it seems like we should be looking at the job data after the `reset_intercept_job_update` bit to show that the job actually stopped with the deleted database error?

Reply:
Excellent call. It turns out this was covering up 2(!) bugs in fabric2_fdb layer.

After updating the test to subscribe and check for db_deleted error, noticed that the job returned as `finished ok` instead of having an error. That was because 

1) We had forgotten to update all the place were DbPrefix was constructed in the recent HCA commit so we still used the `{?DBS, Name}` pattern.

2) There was another bug discovered where we did not reset the  `metadata was checked` before each transaction execution.

Reply:
Nice find!

Should we add the job subscription check to the `handle_db_recreated_when_finished` test as well?

Reply:
Ah, the updated test name makes that more clear. Good call.